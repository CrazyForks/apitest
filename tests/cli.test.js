const { spwanTest } = require("./utils");

describe("cli", () => {
  test("with options --reset", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("start from last failed", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona");
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("with options --ci", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("start from first", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona");
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("with options --only", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona", ["--only", "main.test4"]);
    expect(code).toEqual(0);
    expect(stdout).toMatchSnapshot();
  });
  test("with options --only 2", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona", ["--only", "mod1"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("still start from last failed", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona");
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("with options --dry-run", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona", ["--dry-run"]);
    expect(code).toEqual(0);
    expect(stdout).toMatchSnapshot();
  });
  test("with options --env", async () => {
    const { stdout, code } = await spwanTest("cli/main.jsona", ["--env", "local", "--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("target folder", async () => {
    const { stdout, code } = await spwanTest("cli", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("target folder with options --env", async () => {
    const { stdout, code } = await spwanTest("cli", ["--env", "local", "--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("run group", async () => {
    const { stdout, code } = await spwanTest("cases/run-group.jsona", ["--reset"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("restart failed run group", async () => {
    const { stdout, code } = await spwanTest("cases/run-group.jsona", []);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
});
