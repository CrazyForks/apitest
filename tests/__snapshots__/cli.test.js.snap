// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli restart failed run group 1`] = `
"module main
  fail# retry
    unit test1 ✔
    unit test3 ✘
    main.group3.test3.res.v1: b ≠ a

    {
      \\"req\\": {
        \\"v1\\": \\"a\\"
      },
      \\"res\\": {
        \\"v1\\": \\"a\\"
      }
    }
    unit test1 ✔
    unit test3 ✘
    main.group3.test3.res.v1: b ≠ a

    {
      \\"req\\": {
        \\"v1\\": \\"a\\"
      },
      \\"res\\": {
        \\"v1\\": \\"a\\"
      }
    }
    unit test1 ✔
    unit test3 ✘
    main.group3.test3.res.v1: b ≠ a

    {
      \\"req\\": {
        \\"v1\\": \\"a\\"
      },
      \\"res\\": {
        \\"v1\\": \\"a\\"
      }
    }
"
`;

exports[`cli run group 1`] = `
"module main
  unit vars ✔
  skip ⌾
  delay
    unit test2 ✔
  fail# retry
    unit test1 ✔
    unit test3 ✘
    main.group3.test3.res.v1: b ≠ a

    {
      \\"req\\": {
        \\"v1\\": \\"a\\"
      },
      \\"res\\": {
        \\"v1\\": \\"a\\"
      }
    }
    unit test1 ✔
    unit test3 ✘
    main.group3.test3.res.v1: b ≠ a

    {
      \\"req\\": {
        \\"v1\\": \\"a\\"
      },
      \\"res\\": {
        \\"v1\\": \\"a\\"
      }
    }
    unit test1 ✔
    unit test3 ✘
    main.group3.test3.res.v1: b ≠ a

    {
      \\"req\\": {
        \\"v1\\": \\"a\\"
      },
      \\"res\\": {
        \\"v1\\": \\"a\\"
      }
    }
"
`;

exports[`cli start from first 1`] = `
"module main
  unit test1 ✔
  unit test2 ✘
  main.test2.res.v1: b ≠ a

  {
    \\"req\\": {
      \\"v1\\": \\"a\\"
    },
    \\"res\\": {
      \\"v1\\": \\"a\\"
    }
  }
"
`;

exports[`cli start from last failed 1`] = `
"module main
  unit test2 ✘
  main.test2.res.v1: b ≠ a

  {
    \\"req\\": {
      \\"v1\\": \\"a\\"
    },
    \\"res\\": {
      \\"v1\\": \\"a\\"
    }
  }
"
`;

exports[`cli still start from last failed 1`] = `
"module main
  unit test2 ✘
  main.test2.res.v1: b ≠ a

  {
    \\"req\\": {
      \\"v1\\": \\"a\\"
    },
    \\"res\\": {
      \\"v1\\": \\"a\\"
    }
  }
"
`;

exports[`cli target folder 1`] = `
"module main
  unit test1 ✔
  unit test2 ✘
  main.test2.res.v1: b ≠ a

  {
    \\"req\\": {
      \\"v1\\": \\"a\\"
    },
    \\"res\\": {
      \\"v1\\": \\"a\\"
    }
  }
"
`;

exports[`cli target folder with options --env 1`] = `
"module main
  unit test1 ✔
module mod1
  unit test1 ✘
  mod1.test1.res.v1: b ≠ a

  {
    \\"req\\": {
      \\"v1\\": \\"a\\"
    },
    \\"res\\": {
      \\"v1\\": \\"a\\"
    }
  }
"
`;

exports[`cli with options --ci 1`] = `
"module main
  unit test1 ✔
  unit test2 ✘
  unit test3 ✘
  unit test4 ✔
module mod1
  unit test1 ✘
  unit test2 ✔

1. unit test2(main.test2)
   main.test2.res.v1: b ≠ a

2. unit test3(main.test3)
   main.test3.res.v1: b ≠ a

3. unit test1(mod1.test1)
   mod1.test1.res.v1: b ≠ a

"
`;

exports[`cli with options --dry-run 1`] = `
"module main
  unit test1 (main.test1)
  unit test2 (main.test2)
  unit test3 (main.test3)
  unit test4 (main.test4)
module mod1
  unit test1 (mod1.test1)
  unit test2 (mod1.test2)
"
`;

exports[`cli with options --env 1`] = `
"module main
  unit test1 ✔
module mod1
  unit test1 ✘
  mod1.test1.res.v1: b ≠ a

  {
    \\"req\\": {
      \\"v1\\": \\"a\\"
    },
    \\"res\\": {
      \\"v1\\": \\"a\\"
    }
  }
"
`;

exports[`cli with options --only 1`] = `
"module main
  unit test4 ✔
"
`;

exports[`cli with options --only 2 1`] = `
"module mod1
  unit test1 ✘
  mod1.test1.res.v1: b ≠ a

  {
    \\"req\\": {
      \\"v1\\": \\"a\\"
    },
    \\"res\\": {
      \\"v1\\": \\"a\\"
    }
  }
"
`;

exports[`cli with options --reset 1`] = `
"module main
  unit test1 ✔
  unit test2 ✘
  main.test2.res.v1: b ≠ a

  {
    \\"req\\": {
      \\"v1\\": \\"a\\"
    },
    \\"res\\": {
      \\"v1\\": \\"a\\"
    }
  }
"
`;
